padertorch/data/segment.py

```diff
@@ -0,0 +1,645 @@
+            This is used in _get_segment_length_for_mode
+        padding: May only be `True` if `anchor` is `0` or `left` since padding
+            is only applied to the end of the signal. This may be the right
+            choice for evaluation.
+            If `False` the residual values are disgarded.
+        flatten_separator: specifies the separator used to separate the keys
+            in the flattened dictionary. Defaults to `.`
+    """
+
+    def __init__(self, length: int = -1, shift: int = None,
+                 include_keys: Union[str, list, tuple] = None,
+                 exclude_keys: Union[str, list, tuple] = None,
+                 copy_keys: Union[str, bool, list, tuple] = True,
+                 axis: Union[int, list, tuple, dict] = -1,
+                 anchor: Union[int, str] = 'left',
+                 mode: 'str' = 'constant',
+                 padding: bool = False,
+                 flatten_separator: str = '.'):
+
+        self.include = None if include_keys is None else to_list(include_keys)
+        self.exclude = [] if exclude_keys is None else to_list(exclude_keys)
+        self.length = length
+        if isinstance(axis, (dict, int)):
+            self.axis = axis
+            if isinstance(axis, dict):
+                assert self.include is not None, (self.axis, self.include)
+                assert set(axis.keys()) == set(self.include), (
+                    axis.keys(), self.include
+                )
+        elif isinstance(axis, (tuple, list)):
+            self.axis = to_list(axis)
+            assert self.include is not None, (self.axis, self.include)
+            assert len(axis) == len(include_keys), (
+                'If axis are specified as list it has to have the same length'
+                'as include_keys', axis, include_keys
+            )
+        else:
+            raise TypeError('Unknown type for axis', axis)
+        if shift is None:
+            shift = length
+        # If there is a use case for shift > length, open a pull request and
+        # remove this assert.
+        assert shift <= length, (shift, length)
+        self.shift = shift
+        assert isinstance(anchor, (str, int)), anchor
+        self.anchor = anchor
+        self.copy_keys = to_list(copy_keys)
+        assert all([isinstance(key, (bool, str)) for key in self.copy_keys]), (
+            'All keys in copy_keys have to be str, or copy key has to be one'
+            'boolean', copy_keys
+        )
+        assert mode in possible_segment_modes, (
+            'length_mode has to be one of', possible_segment_modes,
+            'but is', mode
+        )
+        self.mode = mode
+        if padding:
+            # No padding is implemented for the begging of a signal
+            assert anchor in [0, 'left'], (padding, anchor)
+        self.padding = padding
+        self.flatten_separator = flatten_separator
+
+    def __call__(self, example, rng=np.random):
```

Comment:
I'm a fan of type annotations, so can you add `def __call__(self, example: dict, rng=np.random) -> List[dict]`?

Reply:
Also to the other methods?

Reply:
done